seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-fake/src/main/java/org/apache/seatunnel/flink/fake/source/FakeSourceStream.java

```diff
@@ -73,11 +135,23 @@ public String getPluginName() {
 
     @Override
     public void run(SourceFunction.SourceContext<Row> ctx) throws Exception {
-        while (running) {
-            int randomNum = (int) (1 + Math.random() * NAME_ARRAY.length);
-            Row row = Row.of(NAME_ARRAY[randomNum - 1], System.currentTimeMillis());
-            ctx.collect(row);
-            Thread.sleep(TimeUnit.SECONDS.toMillis(1));
+        if (!mockDataEnable) {
+            while (running) {
+                int randomNum = (int) (1 + Math.random() * NAME_ARRAY.length);
+                Row row = Row.of(NAME_ARRAY[randomNum - 1], System.currentTimeMillis());
+                ctx.collect(row);
+                Thread.sleep(TimeUnit.SECONDS.toMillis(1));
+            }
+        } else {
+            int index = 0;
+            while (running && index < mockDataSize){
```

Comment:
This connector is used to mock a stream, I suggest we don't need to add the `mockDataBounded` config here.

Reply:
I have a question, newest version(>1.12) flink api, the batch api is deprecated.
[https://nightlies.apache.org/flink/flink-docs-master/docs/dev/dataset/overview/](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/dataset/overview/).
These can both all in one, Why the stream & batch must be two connector?

Reply:
There exist some deprecated api in ST, we are developing the new ST API. Before our new API finished, it's better to keep the current API.